src/pip/_internal/wheel.py

```diff
@@ -587,25 +578,37 @@ def _get_script_text(entry):
             del console[k]
 
     # Generate the console and GUI entry points specified in the wheel
-    if len(console) > 0:
-        generated_console_scripts = maker.make_multiple(
-            ['%s = %s' % kv for kv in console.items()]
-        )
-        generated.extend(generated_console_scripts)
+    scripts_to_generate.extend(
+        '%s = %s' % kv for kv in console.items()
+    )
+
+    gui_scripts_to_generate = [
+        '%s = %s' % kv for kv in gui.items()
+    ]
+
+    generated_console_scripts = []  # type: List[str]
 
-        if warn_script_location:
-            msg = message_about_scripts_not_on_PATH(generated_console_scripts)
-            if msg is not None:
-                logger.warning(msg)
+    try:
+        generated_console_scripts = maker.make_multiple(scripts_to_generate)
+        generated.extend(generated_console_scripts)
 
-    if len(gui) > 0:
         generated.extend(
-            maker.make_multiple(
-                ['%s = %s' % kv for kv in gui.items()],
-                {'gui': True}
-            )
+            maker.make_multiple(gui_scripts_to_generate, {'gui': True})
+        )
+    except MissingCallableSuffix as e:
+        entry = e.args[0]
+        raise InstallationError(
+            "Invalid script entry point: %s for req: %s - A callable "
+            "suffix is required. Cf https://packaging.python.org/en/"
+            "latest/distributing.html#console-scripts for more "
+            "information." % (entry, req)
```

Comment:
nit: use `.format` in new code please. :)

Reply:
Sure. I'll fix these preemptively in the future, in separate commits from the move or change.

Reply:
@pradyunsg you gave me an idea for a flake8 plugin :)

Reply:
@atugushev I'll be happy to adopt it in pip. :)

Reply:
Addressed [here](https://github.com/pypa/pip/pull/6763/files#diff-79304b3f74b6bf37043da4d96ccd3755R601-R604).

Reply:
@pradyunsg here we go! https://github.com/atugushev/flake8-printf-formatting

```bash
$ flake8 . | wc -l
     297
```